### ISSUE TITLE
How to reproduce the training result of Pose estimation on COCO

### ISSUE THREAD

**Issue by AndywithCV** [ultralytics/ultralytics#6147]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Hello team, 

I have been using YOLOv8 detection and i am happy with it, really appreciate your effor of creating a great yolo version!

I am working on YOLOv8 pose now, and I have some questions:
1. I wanna train YOLOv8 pose from scatch on COCO dataset, is there anything I can do with the training to reproduce the exact result that you guys offer here (the pretrained ), ( what about hyper-parameters like number of epochs, learning rate, optimizer, augmentation and all the stuff specifically), can you provide the default settings yaml file so that I can just use it for training.
2. I wanna to make a new model by modifying the origninal architecture of YOLOv8 pose, then I will compare its performance with the original. Please suggest me any way to compare these 2 model,  other than training these 2 models from scatch on COCO, is there anyway to benchmark them?

Hope to hear your feedback soon!

Thank you so much and have a nice day!

**Comment by glenn-jocher** [comment]
@AndywithCV hello,

I'm glad to hear that you've been finding YOLOv8 useful. The team and the YOLO community put in a lot of effort to make it as great as it is, and your satisfaction is really rewarding for us.

For training YOLOv8 for pose estimation from scratch on COCO dataset, make sure to use the same hyperparameters that were used in the example training found in the pretrained model. The exact parameters such as number of epochs, learning rate, optimizer, augmentation, etc., can vary depending on the specific task at hand, training condition and hardware resource. The default settings that we found optimal could be found in the yaml file available in the repository.

As per your question regarding the comparison of two models, it's important to employ a fair and rigorous approach. You're correct that training both models from scratch on COCO is a good way to compare their performances. However, if you'd like to benchmark the models otherwise, you can use the Benchmark mode of YOLOv8. This will provide you with information about the speed and accuracy of your custom model and the original one, which you can then directly compare. However, do note that this is a complementary method rather than an alternative to training both models on the dataset.

Thank you for your kind words, and we wish you the best in your endeavors with YOLOv8!

**Comment by AndywithCV** [comment]
Thank you for your detailed response, after checking the default yaml file of YOLOv8, i am confused.
This is the default settings from the repository:
https://github.com/ultralytics/ultralytics/blob/main/ultralytics/cfg/default.yaml
in this file, some critical hyperparameters are: number of epochs 100, batch size 16, and Learning rate = 0.01. Do you think that using this default settings will lead to the same performance with the pretrained model? i am using YOLOv8nano version, training on the rtx4070 12gb of vram, what set up do you think that i should adopt, please give me a specific hyperparameter settings.  Besides that is there still any traninig techniques that you used to to achieve that model performance, honestly i am afraid that just by using the default settings I cannot reproduce the model performance from scratch!
Hope to hear from you!!! 
T
hank you so much

**Comment by glenn-jocher** [comment]
@AndywithCV hello,

Thank you for your follow-up question! It's great to see that you are diving deeper into the details of training with YOLOv8.

The default settings provided in the yaml file you mentioned are optimized for an average scenario and act as a good starting point. However, achieving the same performance as the pretrained model from scratch may not be guaranteed with these default settings due to numerous factors including the hardware you are using, the specific dataset characteristics, and more.

Given your hardware (RTX 4070 with 12GB VRAM), the default batch size should be manageable. However, if you find that memory usage is too high, you could reduce it slightly. The learning rate and the number of epochs provided in the default settings can serve as a reasonable starting point.

However, to possibly enhance your results, you might want to make use of additional techniques such as learning rate schedulers, data augmentation, early stopping, etcetera. These techniques can help improve the performance of the model.

Remember, the journey of deep learning involves continuous experimentation and tuning. Feel free to experiment with the hyperparameters and techniques to get the best out of your model.

Keep up the great work, and don't hesitate to ask if you have more questions!

**Comment by AndywithCV** [comment]
I really appreciate your detailed response!

Actually I am not confident of training on COCO-pose from scratch with the same performance as pretrained model, I have an idea of doing the benchmark on Human 3.6m dataset with transfer weight from COCO, do you think that yolov8-pose can be trained on Human3.6m? i have been searching but no one did this yet!!
Looking forward for your reply!

**Comment by glenn-jocher** [comment]
Hello @AndywithCV,

It's great to hear from you again, and I'm glad you're considering diving further into pose estimation with YOLOv8.

Certainly, YOLOv8-pose can be adapted to work with the Human 3.6M dataset. While there may not be precedent for this specific task with YOLOv8, transfer learning is a common approach where you can leverage a model pre-trained on a large dataset like COCO and fine-tune it on a more specific dataset like Human 3.6M.

To proceed, you would start by loading the YOLOv8 model with weights pre-trained on COCO and continue training on the Human 3.6M dataset, making sure to adjust the annotations to match the YOLOv8 expected format. Fine-tuning should help in adapting the knowledge the model has gained from COCO to the nuances of Human 3.6M. 

This approach is popular in practice as it can significantly reduce the needed training time while improving generalization, especially when the target dataset is smaller or more specialized. 

Keep us posted on your progress or if you have further questions. Your exploration and efforts help push the frontiers of what's possible with our models, and we are here to assist you on this journey. Good luck with your training and benchmarking!

Best regards,
Glenn Jocher

**Comment by AndywithCV** [comment]
Thank you so much for your response, really appreciate it!!

**Comment by glenn-jocher** [comment]
Hello @AndywithCV,

You're most welcome, and I'm glad to have been of assistance! It's exciting to hear about your progress and ideas as you explore the capabilities of YOLOv8-pose for your research.

Should you embark on the journey of training on the Human 3.6M dataset using weights transferred from COCO, we'd love to hear about your experience and results. Your pioneering efforts in this area could provide valuable insights for many in the YOLOv8 community.

If you encounter any specific challenges or have further questions as you proceed, please don't hesitate to reach out again. Your inquiries not only help you but also enrich the collective knowledge and resources for all users of YOLOv8-pose.

Wishing you all the best in your modeling endeavors and looking forward to your updates!

Warm regards,
The Ultralytics Team

**Comment by JasseurHadded1** [comment]
@glenn-jocher Can the exact params be given? Thanks

**Comment by glenn-jocher** [comment]
@JasseurHadded1 hello,

For the exact training parameters, please refer to the YOLOv8 documentation. In the Ultralytics Docs, you will find information on default parameter settings that you can use as a baseline for your training on COCO-pose.

**Comment by priyamw** [comment]
> Thank you so much for your response, really appreciate it!!

hey, were you able to train YOLO on Human 3.6m dataset?

**Comment by glenn-jocher** [comment]
Hello @priyamw,

Thank you for your kind words! 😊

Regarding training YOLOv8 on the Human 3.6M dataset, it is indeed feasible. Here are some steps to help you get started:

1. **Dataset Preparation**: Ensure that the Human 3.6M dataset annotations are converted to the YOLO format. Each image should have a corresponding `.txt` file with the format:
    ```
    <class-index> <x> <y> <width> <height> <px1> <py1> <px2> <py2> ... <pxn> <pyn>
    ```
    For 3D keypoints, include the visibility flag:
    ```
    <class-index> <x> <y> <width> <height> <px1> <py1> <p1-visibility> <px2> <py2> <p2-visibility> ... <pxn> <pyn> <pn-visibility>
    ```

2. **YAML Configuration**: Create a dataset YAML file similar to the COCO-Pose example:
    ```yaml
    path: ../datasets/human36m  # dataset root dir
    train: images/train  # train images (relative to 'path')
    val: images/val  # val images (relative to 'path')
    test:  # test images (optional)
    
    # Keypoints
    kpt_shape: [17, 3]  # number of keypoints, number of dims (2 for x,y or 3 for x,y,visible)
    flip_idx: [0, 2, 1, 4, 3, 6, 5, 8, 7, 10, 9, 12, 11, 14, 13, 16, 15]
    
    # Classes dictionary
    names:
      0: person
    ```

3. **Training**: Use the pre-trained COCO weights and fine-tune on the Human 3.6M dataset:
    ```python
    from ultralytics import YOLO

    # Load a pre-trained model
    model = YOLO("yolov8n-pose.pt")

    # Train the model
    results = model.train(data="human36m.yaml", epochs=100, imgsz=640)
    ```

4. **Hyperparameters**: Start with the default hyperparameters and adjust based on your observations. You can find the default settings in the [Ultralytics documentation](https://docs.ultralytics.com).

If you encounter any issues, please provide a minimum reproducible example to help us diagnose the problem. You can find guidance on creating one [here](https://docs.ultralytics.com/help/minimum_reproducible_example).

Feel free to share your progress or any challenges you face. The community and the Ultralytics team are here to support you!

Best of luck with your training!

Warm
